bark/world/map/lane_corridor.hpp

```diff
@@ -53,6 +53,15 @@ struct LaneCorridor {
     return GetLength() - GetS(pt);
   }
 
+  float GetLaneWidth(const Point2d& pt) {
+    uint idx = FindNearestIdx(GetCenterLine(), pt);
+    // assumption: center, left and right have same # elements
+    Point2d left_pt = GetPointAtIdx(GetLeftBoundary(), idx);
```

Comment:
this assumes that these points are always parallel.
A cleaner way would be to calculate the s-value and then query the point.
There is an interpolation available.

Reply:
you mean calculating the s-value of the centerline? and then using this for querying the points for left and right boundary?
I would argue that the s-values of those boundaries are not necessarily corresponding to the s-val of the center line either. Thus, I am not sure it will improve the accuracy of the function, but will certainly make it slower...